src/inet/tests/BUILD.gn

```diff
@@ -63,9 +77,26 @@ chip_test_suite("tests") {
   if (current_os != "mac" && chip_device_platform != "esp32") {
     sources += [
       "TestInetLayerDNS.cpp",
-      "TestInetLayerMulticast.cpp",
       "TestLwIPDNS.cpp",
     ]
     tests += [ "TestInetLayerDNS" ]
   }
 }
+
+# The following binaries should be executed by happy.
+chip_test_suite("happy_tests") {
+  output_name = "libHappyTestInetCommon"
+
+  sources = [ "TestInetLayerMulticast.cpp" ]
+
+  public_configs = [ ":tests_config" ]
+
+  public_deps = [
+    ":chip_inet_test_common_source",
```

Comment:
Change this to

`  ":tests_common"`

and merge `chip_inet_test_common_source` back into `tests`.

I think this should resolve breaking issue with ESP32 tests.